src/netlink_namespace.c

```diff
@@ -0,0 +1,817 @@
+      return MNL_CB_ERROR;
+    }
+
+    kind = mnl_attr_get_str(attr);
+    break;
+  }
+
+  if (kind == NULL) {
+    ERROR("netlink_namespace plugin: qos_filter_cb: kind == NULL");
+    return (-1);
+  }
+
+  { /* The ID */
+    uint32_t numberic_id;
+
+    numberic_id = tm->tcm_handle;
+    if (strcmp(tc_type, "filter") == 0)
+      numberic_id = tm->tcm_parent;
+
+    ssnprintf(tc_inst, sizeof(tc_inst), "%s-%x:%x", kind, numberic_id >> 16,
+              numberic_id & 0x0000FFFF);
+  }
+
+  DEBUG("netlink_namespace plugin: qos_filter_cb: got %s for %s (%i).", tc_type, dev,
+        tm->tcm_ifindex);
+
+  if (check_ignorelist(dev, tc_type, tc_inst))
+    return MNL_CB_OK;
+
+#if HAVE_TCA_STATS2
+  mnl_attr_for_each(attr, nlh, sizeof(*tm)) {
+    struct qos_stats q_stats;
+
+    memset(&q_stats, 0x0, sizeof(q_stats));
+
+    if (mnl_attr_get_type(attr) != TCA_STATS2)
+      continue;
+
+    if (mnl_attr_validate(attr, MNL_TYPE_NESTED) < 0) {
+      ERROR("netlink_namespace plugin: qos_filter_cb: TCA_STATS2 mnl_attr_validate "
+            "failed.");
+      return MNL_CB_ERROR;
+    }
+
+    mnl_attr_parse_nested(attr, qos_attr_cb, &q_stats);
+
+    if (q_stats.bs != NULL || q_stats.qs != NULL) {
+      char type_instance[DATA_MAX_NAME_LEN];
+
+      stats_submitted = 1;
+
+      ssnprintf(type_instance, sizeof(type_instance), "%s-%s", tc_type,
+                tc_inst);
+
+      if (q_stats.bs != NULL) {
+        submit_one(dev, "ipt_bytes", type_instance, q_stats.bs->bytes);
+        submit_one(dev, "ipt_packets", type_instance, q_stats.bs->packets);
+      }
+      if (q_stats.qs != NULL) {
+        submit_one(dev, "if_tx_dropped", type_instance, q_stats.qs->drops);
+      }
+    }
+
+    break;
+  }
+#endif /* TCA_STATS2 */
+
+#if HAVE_TCA_STATS
+  mnl_attr_for_each(attr, nlh, sizeof(*tm)) {
+    struct tc_stats *ts = NULL;
+
+    if (mnl_attr_get_type(attr) != TCA_STATS)
+      continue;
+
+    if (mnl_attr_validate2(attr, MNL_TYPE_UNSPEC, sizeof(*ts)) < 0) {
+      ERROR("netlink_namespace plugin: qos_filter_cb: TCA_STATS mnl_attr_validate2 "
+            "failed.");
+      return MNL_CB_ERROR;
+    }
+    ts = mnl_attr_get_payload(attr);
+
+    if (!stats_submitted && ts != NULL) {
+      char type_instance[DATA_MAX_NAME_LEN];
+
+      ssnprintf(type_instance, sizeof(type_instance), "%s-%s", tc_type,
+                tc_inst);
+
+      submit_one(dev, "ipt_bytes", type_instance, ts->bytes);
+      submit_one(dev, "ipt_packets", type_instance, ts->packets);
+    }
+
+    break;
+  }
+
+#endif /* TCA_STATS */
+
+#if !(HAVE_TCA_STATS && HAVE_TCA_STATS2)
+  DEBUG("netlink_namespace plugin: qos_filter_cb: Have neither TCA_STATS2 nor "
+        "TCA_STATS.");
+#endif
+
+  return MNL_CB_OK;
+} /* int qos_filter_cb */
+
+static int ir_config(const char *key, const char *value) {
+  char *new_val;
+  char *fields[8];
+  int fields_num;
+  int status = 1;
+  char nspath[1024] = "/var/run/netns/";
```

Comment:
I'll move this to the other recommendation you made, which was leaving the path as part of the configuration, instead of haveing it fixed to `/var/run/netns`.